src/services/chain-connector/modules/chain_connector/src/jsonrpc/deal_peer_removed_batched.rs

```diff
@@ -0,0 +1,142 @@
+
+impl DealPeerRemovedReq {
+    fn jsonrpc(&self, host_topic: String, idx: usize) -> JsonRpcReq<GetLogsReq> {
+        let right_boundary = default_right_boundary(&self.left_boundary);
+        let req = GetLogsReq {
+            address: self.deal_id.clone(),
+            topics: vec![DealPeerRemovedData::topic(), host_topic],
+            from_block: self.left_boundary.clone(),
+            to_block: right_boundary,
+        };
+
+        req.to_jsonrpc(idx as u32)
+    }
+}
+
+#[derive(Debug)]
+#[marine]
+pub struct DealPeerRemovedResult {
+    success: bool,
+    error: Vec<String>,
+    is_removed: bool,
+    right_boundary: String,
+    deal_id: String,
+}
+
+impl DealPeerRemovedResult {
+    pub fn ok(right_boundary: String, deal_id: String, is_removed: bool) -> Self {
+        Self {
+            success: true,
+            error: vec![],
+            is_removed,
+            right_boundary,
+            deal_id,
+        }
+    }
+
+    pub fn error(right_boundary: String, deal_id: String, err: String) -> Self {
+        Self {
+            success: false,
+            error: vec![err],
+            is_removed: false,
+            right_boundary,
+            deal_id,
+        }
+    }
+}
+
+#[derive(Debug)]
+#[marine]
+pub struct DealPeerRemovedBatchResult {
+    result: Vec<DealPeerRemovedResult>,
+    success: bool,
+    error: Vec<String>,
+}
+
+impl DealPeerRemovedBatchResult {
+    pub fn empty() -> Self {
+        Self::ok(<_>::default())
+    }
+
+    pub fn ok(result: Vec<DealPeerRemovedResult>) -> Self {
+        Self {
+            success: true,
+            error: vec![],
+            result,
+        }
+    }
+
+    pub fn error(err: String) -> Self {
+        Self {
+            success: false,
+            error: vec![err],
+            result: vec![],
+        }
+    }
+}
+
+fn deal_peer_removed_req_batch(deals: &[DealPeerRemovedReq], host_topic: String) -> Vec<JsonRpcReq<GetLogsReq>> {
+    deals
+        .iter()
+        .enumerate()
+        .map(|(idx, deal)| deal.jsonrpc(host_topic.clone(), idx))
+        .collect::<Vec<_>>()
+}
+
+#[marine]
+pub fn poll_deal_peer_removed_batch(api_endpoint: &str, deals: Vec<DealPeerRemovedReq>) -> DealPeerRemovedBatchResult {
+    if let Err(err) = check_url(api_endpoint) {
+        return DealPeerRemovedBatchResult::error(err.to_string());
+    }
+    if deals.is_empty() {
+        return DealPeerRemovedBatchResult::empty();
+    }
+    let host_topic = match get_encoded_peer_id() {
+        Ok(host_topic) => host_topic,
+        Err(err) => {
+            return DealPeerRemovedBatchResult::error(err.to_string());
+        }
+    };
+    let batch = deal_peer_removed_req_batch(&deals, host_topic);
+    let responses = send_jsonrpc_batch::<GetLogsReq, Vec<Log>>(api_endpoint, batch);
+    let mut responses = match responses {
+        Err(err) => return DealPeerRemovedBatchResult::error(err.to_string()),
+        Ok(r) => r,
+    };
+    responses.sort_by(|a, b| a.id.cmp(&b.id));
```

Comment:
Could you add a comment on why you need to sort?

Reply:
When writing it, I was unsure if JSON-RPC guarantees the order of the responses. I'm still unsure, though.

https://www.jsonrpc.org/specification#batch
> The Response objects being returned from a batch call MAY be returned in any order within the Array. The Client SHOULD match contexts between the set of Request objects and the resulting set of Response objects based on the id member within each Object.

I guess, it's not guaranteed.

Reply:
I guess, we can add the same sort in all batch requests

Reply:
Maybe, in a different PR.

Reply:
Need a comment on why do you need order here at all.